### PR TITLE
Add external parts, remove external-infos

### LINKED ISSUE #267: Create external-properties, external-households, external-tenancies, external-vacancies, remove external-infos

### PR DESCRIPTION
**Pull request by skovmand-wh**
Fixes #267 

I would like feedback on this:

- I'm in doubt which model relations should have `:binary_id` specified, and which should not and whether it even matters at all.
- Can you take a 2nd look at the model fields and verify that the right ones are present?

### Usage of Ecto.Enum

I would like your opinion on this:

I have used `Ecto.Enum` for `ExternalSystem`. This gives us one central definition of external systems in `WiseHomex` that we can use across models, e.g.:

```
field :system, Ecto.Enum, values: WiseHomex.ExternalSystem.values()
```

This also enables consumers to use `WiseHomex.ExternalSystem.values()` inside their applications to e.g. validate field values, to generate dropdowns, etc. For example:

```
def changeset(params) do
  external_systems = WiseHomex.ExternalSystem.values() |> Enum.map(&Atom.to_string/1)

  %__MODULE__{}
  |> cast(params, [:system, :number, :company_number])
  |> validate_required([:system, :number])
  |> validate_inclusion(:system, external_systems, message: "is not supported")
end
```

However, we do not fully support `Ecto.Enum` in JSONParser, and this means that values from the API will not be converted to atoms, but will remain strings in the models, until we change this.

_Update: added this --->_
One concern is if the API sends values that are not supported in a given of WiseHomex. We would need to handle that. For example a user of WiseHomex might not use the newest version. But how to handle that? From the Ecto docs, it looks like attempting to load an invalid enum will simply cause the application to crash: https://hexdocs.pm/ecto/Ecto.Enum.html ... And this would lead to system failures without warning. So we could start using `nil` for invalid values, but then we are going down a path which is not pretty and perhaps also defeats the purpose of `Enum`s in the first place.

I think we should use `:string` for the field instead, and I have changed it to `:string` in https://github.com/wise-home/wise_homex/pull/268/commits/2fa9a7837256655a10d15c841c712c33d5551a2f ... It would make sense though to use Ecto.Enum, if we were the only consumer of `wise_homex` or if it was coupled tighter to an application.

### CODE REVIEW

**Review by lasseebert — CHANGES_REQUESTED** [review]
Very nice and thanks for the thorough writeup on your thoughts.

About `type: :binary_id`
I think it doesn't matter when not using database, in the sense that it will still accept anything.

All the models seem to have the correct set of attributes and relationships :+1: 

I think your findings on Ecto.Enum is totally correct. Even though using it in WiseHomex would be good in the sense that we have more strongly typed fields, it breaks the contract that WiseHomex should be resilient of new (backwards compatible) stuff from the API.

File: lib/wise_homex/models/vacancy_info.ex [hunk 1]
```diff
@@ -7,7 +7,7 @@ defmodule WiseHomex.VacancyInfo do
 
   embedded_schema do
     belongs_to :household, WiseHomex.Household
-    belongs_to :external_info, WiseHomex.ExternalInfo
+    belongs_to :external_vacancy, WiseHomex.ExternalVacancy
```
[the remark this answers is not in the document]
Comment (skovmand-wh) [reply lib/wise_homex/models/vacancy_info.ex]: Good catch :+1: -- fixed in https://github.com/wise-home/wise_homex/pull/268/commits/0262bfdd93b47155b819857f90609cebece01e0e

File: lib/wise_homex/models/external_property.ex [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  embedded_schema do
+    belongs_to :property, WiseHomex.Property
+    has_many :external_households, WiseHomex.ExternalHousehold
+
+    field :system, :string
+    field :number, :integer
+    field :company_number, :integer
+    field :synced, :boolean
+    field :last_synced_at, :utc_datetime
+    field :sync_method, Ecto.Enum, values: [:none, :lurtz, :api]
```
[the remark this answers is not in the document]
Comment (skovmand-wh) [reply lib/wise_homex/models/external_property.ex]: Yes! I forgot that one... Fixed in https://github.com/wise-home/wise_homex/pull/268/commits/4ec992d768cc868546340e64176410878bedde53

**Comment by skovmand-wh** [comment]
@lasseebert Good catches! It should be ready for re-review now.

**Review by lasseebert — APPROVED** [review]
All good 🎉👍

**Pull request merged**